_posts/2024-09-23-wai-aria.md

```diff
@@ -61,15 +61,82 @@ In addition to roles and attributes, there are also rules to consider when using
 
 5. Interactive elements must have an accessible name.
 
-You can find a complete list of rules, with explanations and examples, on W3C's page ["Using ARIA"](https://www.w3.org/TR/using-aria/#firstrule). Some exceptions to these rules are also noted there.
+For a complete list of rules with examples, visit W3C's ["Using ARIA"](https://www.w3.org/TR/using-aria/#firstrule) page. Some exceptions to these rules are also noted there.
 
-I won’t dive into these rules further in this post, as some of them deserve a blog post of their own. Hopefully, more on this topic will come in the future.
+I won’t go into further detail in this post, as some of these rules deserve dedicated blog posts. Hopefully, more on this topic will follow in the future.
+
+## So, When Should You Use ARIA?
+
+Here are a few examples, though the list is far from exhaustive.
+
+### aria-hidden
+
+If a visual element is purely decorative, it should always be marked with `aria-hidden="true"`, letting screen readers and other tools know to ignore it. Even when a logo functions as a link to the homepage, the actual SVG should be marked as `aria-hidden`. I checked the dev blog's logo and the SVG isn't marked as `aria-hidden`. 
+
+```
+<a href="https://www.solita.fi" aria-label="Solita's website">
+  <svg width="46" height="56" xmlns="http://www.w3.org/2000/svg">
+    <path d="M0 56V28h14v28H0zm32-28V0h14v28H32zM16 56V0h14v56H16z" fill="#282828" fill-rule="evenodd"></path>
+  </svg>
+</a> 
+```
+
+The correct way for the SVG:
+
+```
+<svg width="46" height="56" xmlns="http://www.w3.org/2000/svg" aria-hidden="true">
+  <path d="M0 56V28h14v28H0zm32-28V0h14v28H32zM16 56V0h14v56H16z" fill="#282828" fill-rule="evenodd"></path>
+</svg>
+```
+
+Any visual element should either be marked as aria-hidden or given a descriptive, context-aware alt-text.
+
+### aria-valuenow, aria-valuemin, and aria-valuemax
+
+For elements like sliders, users can visually infer the range, but those using assistive technologies require more explicit information. Aria-valuenow indicates the current value, while aria-valuemin and aria-valuemax define the allowable range.
+
+```
+<p id="x-label">How many [things of X] do you want?</p>
+<div
+  role="slider"
+  id="x"
+  aria-valuenow="10"
+  aria-valuemin="0"
+  aria-valuemax="100"
+  aria-labelledby="x-label">
+  ...
+</div>
```

Comment:
Similarly to the note after the next example, here you could also mention that this could be done with native semantics without ARIA like so:
```html
<label for="x">How many [things of X] do you want?</label>
<input type="range" id="x" min="0" max="100" value="10" />
```